src/ChannelServer/World/Character.cs

```diff
@@ -2798,22 +2798,39 @@ public void LookAround()
 			}
 		}
 
-		public void CastSkill(int skillId, float dx, float dy)
+		/// <summary>
+		/// Checks if a player can cast a skill
+		/// </summary>
+		/// <param name="skillId"></param>
+		/// <returns>If skill is castable it returns true</returns>
+		public bool CanCast(int skillId)
 		{
 			// Check skill
 			var skill = this.Skills.Get(skillId);
-			if (skill == null)
-			{
-				//Log.Warning("CZ_SKILL_TARGET_ANI: User '{0}' tried to use skill '{1}', which the character doesn't have.", this.Connection.Account.Name, skillId);
-				Send.ZC_SKILL_CAST_CANCEL(this);
-				//Log.Debug(packet.ToString());
-			}
-			else
+			return CanCast(skill);
+		}
+
+		/// <summary>
+		/// 
+		/// </summary>
+		/// <param name="skill"></param>
+		/// <returns></returns>
+		private bool CanCast(Skill skill)
+		{
+			if (skill != null)
```

Comment:
Little style tip, if you check for null and then return false, you get rid of an indention level, making the code easier to read. The same can be done for the SP check and the require check, until you're left with only 1 or 2 indention levels.